include/triton/core/tritoncache.h

```diff
@@ -0,0 +1,294 @@
+    TRITONCACHE_CacheEntryItem** item);
+
+
+///
+/// CacheEntryItem Lifetime Management
+///
+
+/// Create a new cache entry item object. The caller takes ownership of the
+/// TRITONCACHE_CacheEntryItem object and must call
+/// TRITONCACHE_CacheEntryItemDelete to release the object.
+///
+/// \param item Returns the new cache entry item object.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemNew(
+    TRITONCACHE_CacheEntryItem** item);
+
+/// Delete a cache entry item object.
+///
+/// \param item The cache entry item object to delete.
+/// \return a TRITONSERVER_Error indicating success or failure.
+
+// TODO: Not currently using this API. See TritonCache Lookup/Insert
+// for explanation. Should clean up / clarify the expected ownerships here.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemDelete(
+    TRITONCACHE_CacheEntryItem* item);
+
+///
+/// CacheEntryItem Field Management
+///
+
+// TODO: Proper API descriptions, lifetimes, copies, etc.
+
+// Returns number of buffers held by item
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemBufferCount(
+    TRITONCACHE_CacheEntryItem* item, size_t* count);
+
+// Adds buffer of specified memory_type/id to item. Only CPU memory supported
+// currently.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemAddBuffer(
+    TRITONCACHE_CacheEntryItem* item, const void* base, size_t byte_size,
+    TRITONSERVER_MemoryType memory_type, int64_t memory_type_id);
+
+// Gets buffer at index from item where 0 <= index < count and
+// 'count' is the value returned by TRITONCACHE_CacheEntryItemBufferCount
+// and returns the memory_type info of the buffer
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemGetBuffer(
+    TRITONCACHE_CacheEntryItem* item, size_t index, void** base,
+    size_t* byte_size, TRITONSERVER_MemoryType* memory_type,
+    int64_t* memory_type_id);
+
+///
+/// The following functions can be implemented by a cache. Functions
+/// indicated as required must be implemented or the cache will fail
+/// to load.
+///
+
+/// Create a new cache object.
+///
+/// This function is required to be implemented by the cache.
+///
+/// The caller takes ownership of the
+/// TRITONCACHE_Cache object and must call
+/// TRITONCACHE_CacheDelete to release the object.
+///
+/// This API is implemented by the user-provided cache implementation,
+/// so specific details will be found within the cache implementation.
+///
+/// \param cache Returns the new cache object.
+/// \param config The config options to initialize the cache with.
+///               This will be passed as-is to the cache implementation, so
+///               the expected format and parsing is up to the cache as well.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONCACHE_ISPEC TRITONSERVER_Error* TRITONCACHE_CacheNew(
```

Comment:
How about using `CacheInitialize` and `CacheFinalize` so that it is more similar to the backend API?